### ISSUE TITLE
bookdown table crossrefs work in tufte_html2(), pdf_book() but NOT tufte_handout2()

### ISSUE THREAD

**Issue by maxheld83** [rstudio/bookdown#182]
**TLDR: See hypothesis at the very bottom.**.

I'm adding a simple table like so:

    ```{r desc-example, echo=FALSE}
    kable(x = data.frame(1,2),
          row.names = FALSE,
          caption = "Description Matrix (Subset)")
    ```

I then reference this fantastic table in my `*.rmd` *body*:

```
blah blah as in table \@ref(tab:desc-example)) blah.
```

This works fine, when rendering with `bookdown::tufte_html2()`:

![screen shot 2016-08-21 at 20 22 57](https://cloud.githubusercontent.com/assets/5372770/17839046/1a1c3f08-67dd-11e6-878e-7b2715a56c93.png)

It does *not* work with `bookdown::tufte_handout2()`:

![screen shot 2016-08-21 at 20 24 09](https://cloud.githubusercontent.com/assets/5372770/17839052/3d45e722-67dd-11e6-86e0-7f3d27ae99b1.png)

Astonishingly, it works fine with `bookdown::pdf_book()`, which is *also* LaTeX:

![screen shot 2016-08-21 at 20 25 50](https://cloud.githubusercontent.com/assets/5372770/17839067/7c19bbc2-67dd-11e6-9c75-36a50f3de3c6.png)

*Even weirder*, when looking at the retained `*.tex` produced by `bookdown::tufte_handout2()`, the tex looks just fine:

```
...
blah blah in table \ref{tab:desc-example}) blah
...
\caption{\label{tab:desc-example}Description Matrix (Subset)}
\centering
\begin{tabular}[t]{r|l}
\hline
Index & Description\\
\hline
1 & animals\\
\hline
7 & pronounciation\\
\hline
8 & punctuation\\
\hline
12 & rhymes\\
\hline
\end{tabular}
\end{table}
```

Sure enough, when compiling the tex again from the *tex* source, I get:

![screen shot 2016-08-21 at 20 28 44](https://cloud.githubusercontent.com/assets/5372770/17839086/e7d7ce12-67dd-11e6-84f9-1ccb30d943f1.png)

(though, naturally, the intermediate `*.tex` doesn't work well, because it can't find other stuff as it should, being in a different directory).

----

This problem ... feels ... familiar from working directly in LaTeX ... feels like it wasn't run often enough ... .

**Hypothesis: `bookdown::tufte_handout2()` somehow doesn't run LaTeX often enough for table crossrefs to resolve. `bookdown::pdf_book()` *does* run LaTeX often enough.**.

**Comment by yihui** [comment]
I just tested it, and I think it is a bug of `latexmk`. rmarkdown will use `latexmk` to compile .tex to PDF if it is available, otherwise it will fall back to calling pdflatex/xelatex/bibtex etc. The fallback method works fine, after I removed the latexmk package (`sudo tlmgr remove latexmk` on Mac).
